cockpit/IPC/src/Components/Interface/DbusInterface.tsx

```diff
@@ -50,78 +72,142 @@ export const useDbusInterface = (busName: string, interfaceName: string, objectP
+
+
+
+   useEffect(() => {
+      const checkConnection = setInterval(() => {
+         if (!valid)
+            tryToConnectToDbus();
+      }, 1000);
+
+      return () => {
+         clearInterval(checkConnection);
+      };
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, [valid]);
+
+
+   const retry = <T extends any>(func: () => Promise<T>, retries: number = 3): Promise<T | string> => {
+      return func().catch(async (e: Error) => {
+         if (retries > 0) {
+            console.log(`Retry after error: ${e}`);
+            await new Promise(resolve => setTimeout(resolve, 1000));
+            return retry(func, retries - 1);
+         } else {
+            tryToConnectToDbus();
+            return e.message;
+         }
+      });
+   };
+
+
+
+   const registerSignal = async (signalName: string, signalByte: number): Promise<boolean | string> => {
+      const result = await retry(async () => {
          if (dbusObjectProxy) {
             const response = await dbusObjectProxy.Register_signal(signalName, signalByte);
             console.log("Signal registered successfully: ", response);
             return true;
+         } else {
+            throw new Error("DBus proxy object is null");
          }
-      } catch (error) {
-         console.error("Failed to register signal: ", error);
-      }
-      return false;
+      });
+
+      return typeof result !== 'string';
    };
 
-   const registerSlot = async (slotName: string, slotByte: number): Promise<boolean> => {
-      try {
+   const registerSlot = async (slotName: string, slotByte: number): Promise<boolean | string> => {
+      const result = await retry(async () => {
          if (dbusObjectProxy) {
             const response = await dbusObjectProxy.Register_slot(slotName, slotByte);
             console.log("Slot registered successfully: ", response);
             return true;
+         } else {
+            throw new Error("DBus proxy object is null");
          }
-      } catch (error) {
-         console.error("Failed to register slot: ", error);
-      }
-      return false;
+      });
+
+      return typeof result !== 'string';
    };
 
-   const connect = async (signalName: string, slotName: string): Promise<boolean> => {
-      try {
-         if (dbusObjectProxy !== null && dbusObjectProxy.valid) {
-            await dbusObjectProxy.Connect(slotName, signalName)
+   const connect = async (signalName: string, slotName: string): Promise<boolean | string> => {
+      const result = await retry(async () => {
+         if (dbusObjectProxy) {
+            await dbusObjectProxy.Connect(slotName, signalName);
+            setSlotList((prevSlotList: any) => {
+               let newSlotList = [...prevSlotList];
+               let slot = newSlotList.find(slotItem => slotItem.name === slotName);
+               if (slot) {
+                  slot.connected_to = signalName;
+               }
+               return newSlotList;
```

Comment:
Is this necessary now that we are listening for property changed for all properties?